Change menu

Neodmit opened the pull request:
Change menu to driwer and add styles

Neodmit committed:
Change menu

Change menu to driwer and add styles

Neodmit committed:
menu changes and modal window additions

Change menu code on class witout hooks. Start writing modal window for authorization in settings

Neodmit committed:
Error correction, start working with Settings

Errors were corrected after the review,  layout for settings was written and work with the local storage was started

RomanDashevsky commented on src/components/Home/Home.js:
```diff
@@ -0,0 +1,64 @@
+    render(){
+        return( 
+            <>
+                <IconButton onClick={()=>this.setState({isOpen: true})}> {/*onClick={()=>this.setState({isOpen: true})} */}
```
не коммить закоментированный код

RomanDashevsky replied:
может данный код (MenuButton + Drawer) быть частью Layout?

RomanDashevsky commented on src/components/Home/Home.module.scss:
```diff
@@ -0,0 +1,42 @@
+@import "../../styles/core/import.scss";
```
что за файл import scss? зачем он нужен?

RomanDashevsky commented on src/components/Home/Home.module.scss:
```diff
@@ -0,0 +1,42 @@
+ 
+ .Menu-open{
+    width: 255px !important;
+    background-color: #363740 !important;
```
цвет можно вынести в core

RomanDashevsky commented on src/components/Settings/pics/logo.svg:
```diff
@@ -0,0 +1,10 @@
+<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
```
кажеться этол какой то дублирующий код

RomanDashevsky replied:
все ещё актуально

RomanDashevsky commented on src/hoc/Layout.js:
```diff
@@ -0,0 +1,15 @@
+class Layout extends Component{
+    render(){
+        return(
+            <div>
```
возможно сюда можно добавить общие элементы страницы

RomanDashevsky commented on src/styles/core/import.scss:
```diff
@@ -0,0 +1 @@
+@import "color.scss"
```
переименую файл в index.scss

и утебя будет доступен импорт в духе

@import "../../styles/core";

RomanDashevsky reviewed (changes requested):
глянь комменты

Neodmit committed:
Change after review, working with local storage

I made all the necessary changes after code review, completed the work with the local storage

RomanDashevsky commented on src/App.js:
```diff
@@ -1,27 +1,19 @@
-        />
-        
-      </div>
+        <Layout>
```
Давай все же вынесим Layout в PageComponent

RomanDashevsky replied:
Имею ввиду что давай создадим Пейдж компаненты которые ты будешь использовать в роутинге, и в этой пейдж компаненте будет layout и в нем контент страницы

RomanDashevsky commented on src/components/Home/Home.js:
```diff
@@ -0,0 +1,66 @@
+
+
+class Home extends Component{
+    // constructor(props) {
```
Закомментированный код не коммить

RomanDashevsky commented on src/hoc/Layout.js:
```diff
@@ -0,0 +1,70 @@
+            isOpen: false
+        };
+
+        (window.innerWidth < 479) ? this.variant="" : this.varinat = "permanent"
```
не используй тернарные операторы вне контексте присвоения.
Сделай if/else лучше

RomanDashevsky commented on src/hoc/Layout.js:
```diff
@@ -0,0 +1,70 @@
+function setMenuItems(){
+    return(
+        menuItems.map((menuItems, index) => 
+        <a href={menuItems.link}>
```
не используй в SPA обычные ссылки, посмотри в сторону Link из ReactRouter

RomanDashevsky commented on src/components/Settings/Settings.js:
```diff
@@ -0,0 +1,115 @@
+        return(
+            <>
+                <div className={classes['Login-page']}>
+                    <div className={classes['Login-window']}>
```
попробуй вместо div с кастом класами, использовать Card или Paper компаненты из MaterialUi
Да и в целом в верстке страниц(контента) глянь в сторону секции Layout в MaterialUi: Box/Container/Grid
